app/assets/javascripts/discourse/app/lib/sidebar/user/categories-section/category-section-link.js

```diff
@@ -1,35 +1,93 @@
 import I18n from "I18n";
 
 import { tracked } from "@glimmer/tracking";
+import { get, set } from "@ember/object";
 
 import { bind } from "discourse-common/utils/decorators";
 import Category from "discourse/models/category";
 import { UNREAD_LIST_DESTINATION } from "discourse/controllers/preferences/sidebar";
 
+const DEFAULT_COUNTABLES = [
+  {
+    propertyName: "totalUnread",
+    badgeTextKey: "sidebar.unread_count",
+    route: "discovery.unreadCategory",
+    refreshCountFunction: ({ topicTrackingState, category }) => {
+      return topicTrackingState.countUnread({
+        categoryId: category.id,
+      });
+    },
+  },
+  {
+    propertyName: "totalNew",
+    badgeTextKey: "sidebar.new_count",
+    route: "discovery.newCategory",
+    refreshCountFunction: ({ topicTrackingState, category }) => {
+      return topicTrackingState.countNew({
+        categoryId: category.id,
+      });
+    },
+  },
+];
+
+const customCountables = [];
+
+export function registerCustomCountable({
+  propertyName,
+  badgeTextKey,
+  route,
+  routeQuery,
+  refreshCountFunction,
+  prioritizeOverDefaults,
+}) {
+  customCountables.push({
+    propertyName,
+    badgeTextKey,
+    route,
+    routeQuery,
+    refreshCountFunction,
+    prioritizeOverDefaults,
+  });
+}
+
 export default class CategorySectionLink {
-  @tracked totalUnread = 0;
-  @tracked totalNew = 0;
   @tracked hideCount =
     this.currentUser?.sidebarListDestination !== UNREAD_LIST_DESTINATION;
 
   constructor({ category, topicTrackingState, currentUser }) {
     this.category = category;
     this.topicTrackingState = topicTrackingState;
     this.currentUser = currentUser;
+    this.countables = [...DEFAULT_COUNTABLES];
+
+    if (customCountables.length > 0) {
+      customCountables.forEach((customCountable) => {
+        if (customCountable.shouldRegister({ category })) {
+          if (
+            customCountable?.prioritizeOverDefaults({ category, currentUser })
+          ) {
+            this.countables.unshift(customCountable);
+          } else {
+            this.countables.push(customCountable);
+          }
+        }
+      });
+    }
+
     this.refreshCounts();
   }
 
   @bind
   refreshCounts() {
-    this.totalUnread = this.topicTrackingState.countUnread({
-      categoryId: this.category.id,
-    });
-
-    if (this.totalUnread === 0) {
-      this.totalNew = this.topicTrackingState.countNew({
-        categoryId: this.category.id,
+    this.countables.find((countable) => {
+      const count = countable.refreshCountFunction({
+        topicTrackingState: this.topicTrackingState,
+        category: this.category,
       });
-    }
+
+      set(this, countable.propertyName, count);
```

Comment:
I am not a fan of updating items of an array in a "read" operation such as `find`. This also means that it will update "countable" until the first update and the other could not be updated. Is that intended behavior?

Reply:
Yes this is international because we will only ever display a single count. Therefore, there is no need for us to update the rest.